packages/api/cms-api/src/action-logs/action-logs-context.plugin.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from "@nestjs/common";
+import { Reflector } from "@nestjs/core";
+import { GqlExecutionContext } from "@nestjs/graphql";
+import { ClsPluginBase } from "nestjs-cls";
+
+import { DISABLE_COMET_GUARDS_METADATA_KEY } from "../auth/decorators/disable-comet-guards.decorator";
+import { User } from "../user-permissions/interfaces/user";
+import { ActionLogsContextModule } from "./action-logs-context.module";
+import { ActionLogsContextService } from "./action-logs-context.service";
+
+@Injectable()
+export class ActionLogsContextPlugin extends ClsPluginBase {
+    constructor() {
+        super("action-logs-context-plugin");
+
+        this.registerHooks({
+            inject: [Reflector, ActionLogsContextService],
+            useFactory: (reflector: Reflector, service: ActionLogsContextService) => ({
+                afterSetup: async (_cls, initContext) => {
+                    if (initContext.kind !== "interceptor" || service.hasUserId()) return;
+
+                    const { ctx } = initContext;
+                    const request: Request & { user: User } =
+                        ctx.getType().toString() === "graphql" ? GqlExecutionContext.create(ctx).getContext().req : ctx.switchToHttp().getRequest();
+                    const disableCometGuard = reflector.getAllAndOverride(DISABLE_COMET_GUARDS_METADATA_KEY, [ctx.getHandler(), ctx.getClass()]);
+                    if (disableCometGuard) return;
+
+                    await service.setUserId(request.user.id);
+                },
+            }),
+        });
+
+        this.imports.push(ActionLogsContextModule);
```

Comment:
Why do we need this import? The import is also added in `ActionLogsModule.forRoot()`.

Reply:
I dont wanted to make the whole module global because this is an anti pattern for me, so it has to be imported here because the plugin is registered in nestjs-cls

Reply:
nope we need the service and I also think this is no magic this is like the docs explains and nestjs is working, magic would be when make it globally available and use it directly